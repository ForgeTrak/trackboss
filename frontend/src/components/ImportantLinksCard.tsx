import React from 'react';
import { Box, Center, ListItem, Link, UnorderedList, Heading, VStack, Divider } from '@chakra-ui/react';

export default function ImportantLinksCard() {
    return (
        <Box bg="white" boxShadow="md" border="1px" borderColor="gray.200" p={3}>
            <Center>
                <VStack boxSize="md">
                    <Heading>Important Links</Heading>
                    <Divider />
                    <UnorderedList pt={10}>
                        <Link
                            href="https://palmyramx.com/pages/track-rules"
                            target="_blank"
                            fontSize="4xl"
                            color="orange"
                        >
                            <ListItem pt={1}>Member Rules</ListItem>
                        </Link>
                        <Link
                            href="https://twitter.com/palmyramx"
                            target="_blank"
                            fontSize="4xl"
                            color="orange"
                        >
                            <ListItem pt={1}>Twitter</ListItem>
                        </Link>
                        <Link
                            href="https://www.facebook.com/palmyramx/"
                            target="_blank"
                            fontSize="4xl"
                            color="orange"
                        >
                            <ListItem pt={1}>Facebook</ListItem>
                        </Link>
                        <Link
                            href="https://www.instagram.com/palmyramx_hogback_hill/?hl=en"
                            target="_blank"
                            fontSize="4xl"
                            color="orange"
                        >
                            <ListItem pt={1}>Instagram</ListItem>
                        </Link>
                    </UnorderedList>
                </VStack>

            </Center>
        </Box>
    );
}
