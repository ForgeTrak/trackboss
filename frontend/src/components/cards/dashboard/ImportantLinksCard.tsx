/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { Box, ListItem, Link, UnorderedList, Heading, VStack, Divider, Center } from '@chakra-ui/react';
import { Link as DashboardLink } from '../../../../../src/typedefs/link';

interface cardProps {
    dashboardLinks: DashboardLink[],
    memberId: number,
    jwt: string,
}

export default function ImportantLinksCard(props: cardProps) {
    return (
        <Box bg="white" boxShadow="md" border="1px" borderColor="gray.200" p={2} h="375" m={2}>
            <VStack boxSize="md" alignItems="left" ml={4}>
                <Center>
                    <Heading size="lg">Club Info and Links</Heading>
                </Center>
                <Divider />
                <UnorderedList>
                    <Link
                        target="_blank"
                        fontSize="2xl"
                        // eslint-disable-next-line max-len
                        href={`${process.env.REACT_APP_API_URL}/api/member/card/create/${props.memberId}?id=${props.jwt}`}
                        color="orange"
                    >
                        <ListItem>
                            Membership Card
                        </ListItem>
                    </Link>
                    {
                        // eslint-disable-next-line arrow-body-style
                        props.dashboardLinks.map((link: DashboardLink) => {
                            const displayLink = (
                                <Link
                                    href={link.linkUrl}
                                    target="_blank"
                                    fontSize="2xl"
                                    color="orange"
                                >
                                    <ListItem pt={1}>{link.linkTitle}</ListItem>
                                </Link>
                            );
                            return displayLink;
                        })
                    }
                </UnorderedList>
            </VStack>
        </Box>
    );
}
