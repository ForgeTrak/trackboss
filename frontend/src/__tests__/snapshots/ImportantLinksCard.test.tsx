import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ImportantLinksCard from '../../components/ImportantLinksCard';

it('header renders correctly', () => {
    render(
        <BrowserRouter>
            <ImportantLinksCard />
        </BrowserRouter>,
    );
    expect(screen.getByText('Important Links')).toBeInTheDocument();
});

it('member rules link renders correctly', () => {
    render(
        <BrowserRouter>
            <ImportantLinksCard />
        </BrowserRouter>,
    );
    expect(screen.getByText('Member Rules').closest('a'))
        .toHaveAttribute('href', 'https://palmyramx.com/pages/track-rules');
});

it('palmyra racing twitter link renders correctly', () => {
    render(
        <BrowserRouter>
            <ImportantLinksCard />
        </BrowserRouter>,
    );
    expect(screen.getByText('Twitter').closest('a'))
        .toHaveAttribute('href', 'https://twitter.com/palmyramx');
});

it('palmyra racing facebook link renders correctly', () => {
    render(
        <BrowserRouter>
            <ImportantLinksCard />
        </BrowserRouter>,
    );
    expect(screen.getByText('Facebook').closest('a'))
        .toHaveAttribute('href', 'https://www.facebook.com/palmyramx/');
});

it('palmyra racing instagram link renders correctly', () => {
    render(
        <BrowserRouter>
            <ImportantLinksCard />
        </BrowserRouter>,
    );
    expect(screen.getByText('Instagram').closest('a'))
        .toHaveAttribute('href', 'https://www.instagram.com/palmyramx_hogback_hill/?hl=en');
});
