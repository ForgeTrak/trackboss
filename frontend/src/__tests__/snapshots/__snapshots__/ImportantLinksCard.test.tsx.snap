// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-1nfevod"
>
  <div
    className="css-gmuwbf"
  >
    <div
      className="chakra-stack css-ef3ona"
    >
      <h2
        className="chakra-heading css-0"
      >
        Important Links
      </h2>
      <hr
        aria-orientation="horizontal"
        className="chakra-divider css-1upb9tn"
      />
      <ul
        className="css-t2v3d8"
        role="list"
      >
        <a
          className="chakra-link css-1ffay1v"
          href="https://palmyramx.com/pages/track-rules"
          target="_blank"
        >
          <li
            className="css-18csb69"
          >
            Member Rules
          </li>
        </a>
        <a
          className="chakra-link css-1ffay1v"
          href="https://twitter.com/palmyramx"
          target="_blank"
        >
          <li
            className="css-18csb69"
          >
            Twitter
          </li>
        </a>
        <a
          className="chakra-link css-1ffay1v"
          href="https://www.facebook.com/palmyramx/"
          target="_blank"
        >
          <li
            className="css-18csb69"
          >
            Facebook
          </li>
        </a>
        <a
          className="chakra-link css-1ffay1v"
          href="https://www.instagram.com/palmyramx_hogback_hill/?hl=en"
          target="_blank"
        >
          <li
            className="css-18csb69"
          >
            Instagram
          </li>
        </a>
      </ul>
    </div>
  </div>
</div>
`;
